### PR TITLE
Correct parsing when called as `rhts`

### PR DESCRIPTION
**Pull request by pfdaly**
To correct issue raised in [tmt #1353 ](https://github.com/teemtee/tmt/issues/1353)

[tmt_1353_test_results.txt](https://github.com/teemtee/tmt/files/10094528/tmt_1353_test_results.txt)

### CODE REVIEW

File: tmt/steps/execute/scripts/tmt-report-result [hunk 1]
```diff
@@ -11,6 +11,61 @@ help=False
...
+        fi
+        exit 0
+    else
+        # Create the output file with the report contents.
+        write_report_file
+        exit 0
+    fi
+}
+
+if [[ "$0" == *"rhts-report-result"* ]]; then
```
Comment (lukaszachy) [inline tmt/steps/execute/scripts/tmt-report-result:61]: Does this match in some configuration? I've tried provided reproducer but `$0` had the value `/usr/local/bin/rhts-report-result` during the execution.

File: tmt/steps/execute/scripts/tmt-report-result [hunk 2]
```diff
@@ -11,6 +11,61 @@ help=False
...
+    TESTNAME=$1
+    TESTRESULT=$2
+    outputFile=$3
+    METRIC=$4
+    check_report_file
```
Comment (lukaszachy) [inline tmt/steps/execute/scripts/tmt-report-result:66]: Don't we call check_report_file twice? The final one (line 137) is executed as well.

**Review by lukaszachy — COMMENTED** [review]
Sorry for being late. Two questions raised.

File: tmt/steps/execute/scripts/tmt-report-result (hunk 1 above)
Comment (pfdaly) [reply tmt/steps/execute/scripts/tmt-report-result:61]: The '*' wildcard allows the matching of all characters before the 'rhts-report-result' string, and as such it matches with the command regardless of the origin directory.

> [root@fedora scripts]# /usr/local/bin/rhts-report-result /smoke_test PASS rhts_report 69
> [root@fedora scripts]# cat /tmp/restraint-result 
> SERVER=
> PORT=
> MESSAGE=
> OUTPUTFILE=rhts_report
> DISABLEPLUGIN=
> TESTNAME=/smoke_test
> TESTRESULT=PASS
> METRIC=69
> [root@fedora scripts]#

> [root@fedora scripts]# rhts-report-result /second_smoke_test PASS rhts_report 75
> [root@fedora scripts]# cat /tmp/restraint-result 
> SERVER=
> PORT=
> MESSAGE=
> OUTPUTFILE=rhts_report
> DISABLEPLUGIN=
> TESTNAME=/second_smoke_test
> TESTRESULT=PASS
> METRIC=75
> [root@fedora scripts]#

File: tmt/steps/execute/scripts/tmt-report-result (hunk 2 above)
Comment (pfdaly) [reply tmt/steps/execute/scripts/tmt-report-result:66]: The function 'check_report_file' results with the script exiting.

This BASH script deals with two paths: rhts & restraint. rhts requires less processing, and so the 'check_report_file' function can be called earlier to exit execution, whereas restraint requires the script to continue on through the additional logic of the 'check_opt_args' function before finally executing the 'check_report_file' function to exit. If the rhts path does not call the 'check_report_file' function then the script continues to execute and the variable assignment for the contents in the 'restraint-result' file end up incorrect.

**Comment by thrix** [comment]
@psss @pfdaly @lukaszachy what about this leftover?

**Comment by thrix** [comment]
Adding to 1.29, this is a Kernel QE blocker https://issues.redhat.com/browse/KQE-18

**Comment by pfdaly** [comment]
I thought this had gone through. It must've fallen through the cracks when I had to shift my focus to the ER CTC testing for automotive.

I don't believe there was anything remaining with this.

I had checked that the new code corrected the original issue with the values being assigned to the incorrect variables in the generated output file when called with rhts-report-result. I also confirmed that the change did not break the previous behavior, such that when called with rstrnt-report-result the values are assigned to the correct variables.

```
[root@ibm-p8-kvm-03-guest-02 ~]# rhts-report-result /smoke_test PASS rhts_report 69
[root@ibm-p8-kvm-03-guest-02 ~]# cat /tmp/restraint-result 
SERVER=
PORT=
MESSAGE=
OUTPUTFILE=rhts_report
DISABLEPLUGIN=
TESTNAME=/smoke_test
TESTRESULT=PASS
METRIC=69
[root@ibm-p8-kvm-03-guest-02 ~]#



[root@ibm-p8-kvm-03-guest-02 ~]# rstrnt-report-result --server http://test-example.com --port 55 --disable-plugin avc --message 'Example output message.' -o /tmp/example_output.txt report PASS 66
[root@ibm-p8-kvm-03-guest-02 ~]# cat /tmp/restraint-result 
SERVER=http://test-example.com
PORT=55
MESSAGE=Example output message.
OUTPUTFILE=/tmp/example_output.txt
DISABLEPLUGIN=avc
TESTNAME=report
TESTRESULT=PASS
METRIC=66
[root@ibm-p8-kvm-03-guest-02 ~]# 
```

**Comment by lukaszachy** [comment]
@pfdaly Could you create a test case for this? E.g. what you used to verify this pr fixes the issue.

**Comment by pfdaly** [comment]
Hi @lukaszachy , I've added tests to check both the results of calling rhts-report-result individually and multiple times. This ensures that the output file is populated correctly, and that the result hierarchy is honoured.

**Review by lukaszachy — APPROVED** [review]

**Comment by pfdaly** [comment]
I can't say that I recall how the incorrect value was being read, given the time that's passed since implementation. Nor can I reproduce the initial failure on the latest tmt. I'd have to try to install tmt version 1.15.

The initial issue was that when the report script was called multiple times within a test that the result hierarchy, (Fail > Warn > Pass > Skip), was not being honoured, and so a result of PASS was being returned instead of WARN. The file I attached, 'tmt_1353_test_results.txt', to the initial comment shows the test which juk supplied executing as expected post fix implementation, and also the test which juk supplied is what I've included in this commit.

The issue you've highlighted, with two results per test, is something I posted about in the tmt slack channel last week, after I upgraded from 1.25 to 1.28, and was opened as [2389](https://github.com/teemtee/tmt/issues/2389) . The first result is derived from the contents of the report-result file, while the second result is derived from the return code of the tests bash script. This can about after the following [commit](https://github.com/teemtee/tmt/commit/ece9db96caa8611f87f43e6e221e9ab793237b01#diff-458194650925d6c3ea27ee8a599db67a78a23dbb0b353cddc9e2b8387b54d993R404)

**Comment by psss** [comment]
Ok, I see, so we are handling multiple problems here. What about moving the responsibility of result hierarchy handling to python code? See also my [comment](https://github.com/teemtee/tmt/issues/2389#issuecomment-1775383556) in the phase results issue which should be covered by the approach as well.

**Review by psss — APPROVED** [review]
Ok, let's merge this partial fix as the first step and continue with the result deduplication in #2389.

**Pull request merged**